extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/OTelExporterRecorder.java

```diff
@@ -145,18 +175,135 @@ private SpanExporter createHttpSpanExporter(OtlpExporterRuntimeConfig exporterRu
+                                                vertx.get()),
+                                        MeterProvider::noop,
+                                        exportAsJson),
+                                aggregationTemporalityResolver(metricsConfig),
+                                aggregationResolver(metricsConfig));
+                    } else {
+                        throw new IllegalArgumentException(String.format("Unsupported OTLP protocol %s specified. " +
+                                "Please check `quarkus.otel.exporter.otlp.traces.protocol` property", protocol));
+                    }
+
+                } catch (IllegalArgumentException iae) {
+                    throw new IllegalStateException("Unable to install OTLP Exporter", iae);
+                }
+                return metricExporter;
+            }
+        };
+    }
+
+    private static DefaultAggregationSelector aggregationResolver(OtlpExporterMetricsConfig metricsConfig) {
+        String defaultHistogramAggregation = metricsConfig.defaultHistogramAggregation()
+                .map(s -> s.toLowerCase(Locale.ROOT))
+                .orElse("explicit_bucket_histogram");
+
+        DefaultAggregationSelector aggregationSelector;
+        if (defaultHistogramAggregation.equals("explicit_bucket_histogram")) {
+            aggregationSelector = DefaultAggregationSelector.getDefault();
+        } else if (BASE2EXPONENTIAL_AGGREGATION_NAME.equalsIgnoreCase(defaultHistogramAggregation)) {
+
+            aggregationSelector = DefaultAggregationSelector
+                    .getDefault()
+                    .with(InstrumentType.HISTOGRAM, Aggregation.base2ExponentialBucketHistogram());
+
+        } else {
+            throw new ConfigurationException(
+                    "Unrecognized default histogram aggregation: " + defaultHistogramAggregation);
+        }
+        return aggregationSelector;
+    }
+
+    private static AggregationTemporalitySelector aggregationTemporalityResolver(OtlpExporterMetricsConfig metricsConfig) {
+        String temporalityValue = metricsConfig.temporalityPreference()
+                .map(s -> s.toLowerCase(Locale.ROOT))
+                .orElse("cumulative");
+        AggregationTemporalitySelector temporalitySelector;
+        switch (temporalityValue) {
+            case "cumulative":
+                temporalitySelector = AggregationTemporalitySelector.alwaysCumulative();
+                break;
+            case "delta":
+                temporalitySelector = AggregationTemporalitySelector.deltaPreferred();
+                break;
+            case "lowmemory":
+                temporalitySelector = AggregationTemporalitySelector.lowMemory();
+                break;
+            default:
+                throw new ConfigurationException("Unrecognized aggregation temporality: " + temporalityValue);
+        }
+        return temporalitySelector;
```

Comment:
Nitpick: I am pretty sure this can be improved with the "new" enhanced switch

Reply:
done